src/base/settingsstorage.h

```diff
@@ -62,4 +62,27 @@ public slots:
     mutable QReadWriteLock m_lock;
 };
 
+template<typename T>
+class SettingsStorageHelper
+{
+public:
+    SettingsStorageHelper(const QString &keyName)
+        : m_keyName(keyName)
+    {
+    }
+
+    void set(const T &value)
+    {
+        SettingsStorage::instance()->storeValue(m_keyName, value);
+    }
+
+    T get(const QVariant &defaultValue = QVariant())
+    {
+        return SettingsStorage::instance()->loadValue(m_keyName, defaultValue).template value<T>();
+    }
+
+private:
+    QString m_keyName;
```

Comment:
```C++
template <class T>
class A {
public:
	A(const QLatin1String& s)
		: s_(s) {
	}
 	A(const char* s)
		: s_(s) {
	}
private:
	QLatin1String s_;
};

A<int> a(QLatin1String("dgfdg"));
A<int> b("dgfdg");
```

works here. But of course its dangerous, because `Qlatin1String` does not copy the data.